lib/msal-node/src/client/ClientCredentialClient.ts

```diff
@@ -214,23 +222,41 @@ export class ClientCredentialClient extends BaseClient {
                 thumbprint
             );
 
-            // check if a new token is being requested because the cached token needs to be refreshed, not because it's expired
-            if (accessTokenCacheKey) {
-                // check if ESTS is down (status code 429 or 5xx)
-                if ((response.status === 429) || (response.status >= 500)) {
+            serverTokenResponse = response.body;
+
+            // if a new token is being requested due to the refresh_in value
+            if (accessTokenRefresh) {
```

Comment:
As stated before, the logic should not be here. It should be in `acquireToken`. This method does too much and at the same time it doesn't cover errors from block at L174